### DISCUSSION TITLE
Define collections and relationships with code

### DISCUSSION THREAD

**Discussion by VaZark** [directus/directus#11456] [Feature Requests]
It would be pretty cool if we could extract the DB Schema with `npx directus schema generate` to create something similar to Models used by SqlAlchemy or Django ORM. (I admit to being a bit biased having worked with Django in the past)

Use cases:

1. Easier to write code for extensions as it can help to have less mental overhead when dealing working with data from the DB as they have types. When used with TS, it will also help avoid a whole class of errors
2. Can be used for autogenerating DB schema changes that can be used to promote them from dev to production in a consistent manner by reducing human error.

**Comment by VaZark** [comment]
There seems to be a diffing logic for schema using [umzug](https://github.com/sequelize/umzug) used by MikroORM.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
